core/debug/src/debug.c

```diff
@@ -37,27 +37,40 @@
  * overwritten to provide a different mechanism for printing messages (e.g. the
  * UART port instead of semihosting). */
 #define DEBUG_MAX_BUFFER 128
-uint8_t g_buffer[DEBUG_MAX_BUFFER];
-int g_buffer_pos = 0;
+static uint8_t g_buffer[DEBUG_MAX_BUFFER];
+static uint32_t g_buffer_pos = 0;
+
+__attribute__((section(".uninitialized"))) static uint32_t g_semihosting_magic;
+
+/* This method is an api method which initialize the g_semihosting_magic variable,
+ * when debugging this api should be called through gdb
+ */
+void debug_semihosting_enable(void)
```

Comment:
`extern` declare this function in a header file to avoid forward declaring it manually later on.